src/cargo/core/workspace.rs

```diff
@@ -647,6 +651,14 @@ impl<'gctx> Workspace<'gctx> {
         self
     }
 
+    pub fn requested_lockfile_path(&self) -> Option<&PathBuf> {
+        self.requested_lockfile_path.as_ref()
+    }
+
+    pub fn set_requested_lockfile_path(&mut self, path: Option<PathBuf>) {
+        self.requested_lockfile_path = path;
+    }
```

Comment:
Could we make `lockfile_path` work in the similar pattern like `target_dir`? That is,

* `pub fn lockfile_path()` returns a  `FileSystem`. By default check `Workspace.requested_lockfile_path`, which is `Option<FileSystem>`.
* If none, calls `fn default_lockfile_path()` to get `ws.root.join("Cargo.lock")`.

Reply:
So to clarify, we basically want to move logic on of `cargo::ops::lockfile::lock_root` into worspace instead, right? I was originally thinking about it, but wasn't sure if I should move that piece of code around